lib/lifeguard/infinite_threadpool.rb

```diff
@@ -27,8 +32,12 @@ def check_queued_jobs
       return if busy?
       return if @queued_jobs.size <= 0
 
-      queued_job = @queued_jobs.pop
-      async(*queued_job[:args], &queued_job[:block])
+      @super_async_mutex.synchronize do
+        return if busy?
+        queued_job = @queued_jobs.pop
+        super_async(*queued_job[:args], &queued_job[:block])
+      end
+
       check_queued_jobs
```

Comment:
I still think this recursive call is a problem. A loop would ensure we never blow the stack.